extensions/MozChangeField/lib/Post/SeverityS1PriorityP1.pm

```diff
@@ -0,0 +1,44 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# This Source Code Form is "Incompatible With Secondary Licenses", as
+# defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Extension::MozChangeField::Post::SeverityS1PriorityP1;
+
+use 5.10.1;
+use Moo;
+
+use Bugzilla::Field;
+
+sub evaluate_create {
+  my ($self, $args) = @_;
+  my $bug       = $args->{bug};
+  my $timestamp = $args->{timestamp};
+
+  if ($bug->bug_severity eq 'S1' && $bug->priority ne 'P1' && grep { $_ eq 'P1' }
+    @{get_legal_field_values('priority')})
```

Comment:
`grep` in a boolean context is inefficient; use `any` (see `PerlCritic/ProhibitBooleanGrep`)